### ISSUE TITLE
Settlement Route with Test Transactions

### ISSUE THREAD

**Issue by ooiginla** [braintree/braintree_php#327]
### General information

* SDK/Library version: 6.9.0
* Environment: Sandbox
* Language, language version, and OS: PHP on Ubuntu 20.6 -->

### Issue description

In the file Braintree\TestingGateway\Transaction::_doTestRequest() method. 

```
 private function _doTestRequest($testPath, $transactionId)
    {
        self::_checkEnvironment();
        $path = $this->_config->merchantPath() . '/transactions/' . $transactionId . $testPath;
        $response = $this->_http->put($path); 
        return Transaction::factory($response['transaction']);
    }
```
When you try to settle a transaction that has been previously settled, it gives an error undefined parameter ($response['transaction']). This can be properly handled.

```
^ ErrorException {#1051 ▼
  #message: "Undefined array key "transaction""
  #code: 0
  #file: "/var/www/html/vendor/braintree/braintree_php/lib/Braintree/TestingGateway.php"
  #line: 44
  #severity: E_WARNING
  trace: {...}
}
```

**Comment by hollabaq86** [comment]
👋 @ooiginla thanks for reaching out. Can I get a little more info on your use case, here? This function is only available in the [TestingGateway](https://github.com/braintree/braintree_php/blob/e35c2a2d74a10e9cec5932506486e4b8bccfa4ba/lib/Braintree/TestingGateway.php#L21) class, which is used in a handful of tests that are designed to be run once. Why are you using the TestGateway?

**Comment by ooiginla** [comment]
The documentation under testing settlement routes states we can use this method to move transactions from submitted for settlement to settled and the escrow status to held during testing. We have exposed this api to other product clients within our ecosystem but when a transaction is already settled in test, this method throws an error because it is expecting a transaction and does hot handle that use case where it returns the full error object.

**Comment by hollabaq86** [comment]
>The documentation under testing settlement routes

What documentation? Can you provide me a link? I'm trying to get a full understanding of what we need to be changing.

The [preferred way](https://developer.paypal.com/braintree/docs/reference/request/transaction/submit-for-settlement/php) to test settling a transaction is via [Transaction->submitForSettlement()](https://github.com/braintree/braintree_php/blob/master/lib/Braintree/TransactionGateway.php#L562) using sandbox credentials.

**Comment by ooiginla** [comment]
So when I am creating a transaction, I already have submitted_for_settlement option as true. So, my transaction comes back as status: **submitted_for_settlement** and escrow_status: **hold_pending**

https://developer.paypal.com/braintree/docs/reference/general/testing/php#settlement-status

```
$sale_result = $gateway->transaction()->sale([
  'amount' => '100.00',
  'paymentMethodNonce' => nonceFromTheClient,
  'options' => [
    'submitForSettlement' => True
  ]
]);

// This is why i am calling the settle method on the test
$transaction = $gateway->testing()->settle($sale_result->transaction->id);
get_class($transaction)
# Braintree\Transaction

$transaction->status
# Braintree\Transaction::SETTLED
```

If you trace the $gateway->testing()->settle() method, it calls behind the scenes
 TestingGateway::settle() method which returns  self::_doTestRequest('/settle', $transactionId);
 
 hence the reason i started with that in my first complaint, the doTestRequests returns Transaction::factory($response['transaction']);  
 
 In a case i am passing an already settled transaction to 
  $gateway->testing()->settle($sale_result->transaction->id);  
  
  it throws a bug. i hope this is clear.

**Comment by hollabaq86** [comment]
Thanks @ooiginla the documentation link and your use case really helped clear things up.

Anyhoo, confusion cleared and we'll look into fixing this up. If you want to PR the fix, feel free, but don't feel obligated to!

For internal tracking, issue 7065

**Comment by hollabaq86** [comment]
👋 @ooiginla we released a fix for this in v6.11.2, let us know if you still encounter issues!
